File: problem_002.swift
#!/usr/bin/xcrun swift
// -*- compile-command: "swiftc problem_001.swift -o problem_001" -*-
// Copyright (c) 2016 Michael Caldwell

print("Project Euler - Problem 2:")
print("By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.\n")

let limit = 4000000
var total = 0

var a = 1
var b = 1
var c = 1

while a < limit {
    a = b + c
    c = b
    b = a
    if b % 2 == 0 {
        total += b
    }
}

print(total)

File: problem_002.js
#!/usr/bin/node

console.log("Project Euler - Problem 2:");
console.log("By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.\n");

const limit = 4000000;
var sum = 0;

let f1 = 1;
let f2 = 0;
while (f1 < limit){
    let t = f1 + f2;
    f2 = f1;
    f1 = t;

    if (f1 % 2 == 0)
        sum += f1;
}

console.log(sum);
